.github/workflows/build.yml

```diff
@@ -0,0 +1,172 @@
+    steps:
+      - uses: actions/checkout@v2
+        name: Clone
+      - uses: actions/setup-node@v1
+        name: Setup Node
+        with:
+          node-version: "*"
+      - name: Install Yarn
+        run: corepack enable
+      - name: Install npm dependencies
+        run: yarn install
+      - name: Build
+        run: yarn run build
+      - uses: actions/upload-artifact@v3
+        name: Upload Artifacts
+        with:
+          name: tauri-web
+          path: dist/*
+  build-linux:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        name: Clone
+      - uses: actions-rs/toolchain@v1
+        name: Setup Rust
+        with:
+          toolchain: stable
+          override: true
+      - uses: actions/setup-node@v1
+        name: Setup Node
+        with:
+          node-version: "*"
+      - name: Install Yarn
+        run: corepack enable
+      - name: Install npm dependencies
+        run: yarn install
+      - name: Install tauri dependencies
+        run: sudo apt update && sudo apt install -y build-essential libpython3-dev libdbus-1-dev libgtk-3-dev libsoup-gnome2.4-dev libwebkit2gtk-4.0-dev
+      - name: Build Tauri
+        run: yarn run build:tauri
+      - uses: actions/upload-artifact@v3
+        name: Upload Artifacts (appimage)
+        with:
+          name: tauri-linux
+          path: src-tauri/target/release/bundle/appimage/*.AppImage
+      - uses: actions/upload-artifact@v3
+        name: Upload Artifacts (deb)
+        with:
+          name: tauri-debian
+          path: src-tauri/target/release/bundle/deb/*.deb
+  build-windows:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v2
+        name: Clone
+      - uses: actions-rs/toolchain@v1
+        name: Setup Rust
+        with:
+          toolchain: stable
+          override: true
+      - uses: actions/setup-node@v1
+        name: Setup Node
+        with:
+          node-version: "*"
+      - name: Install Yarn
+        run: corepack enable
+      - name: Install npm dependencies
+        run: yarn install
+      - name: Build Tauri
+        run: yarn run build:tauri
+      - uses: actions/upload-artifact@v2
+        name: Upload Artifacts
+        with:
+          name: tauri-windows
+          path: src-tauri/target/release/bundle/msi/*.msi
+  build-macos:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v2
+        name: Clone
+      - uses: actions-rs/toolchain@v1
+        name: Setup Rust
+        with:
+          toolchain: stable
+          override: true
+      - uses: actions/setup-node@v1
+        name: Setup Node
+        with:
+          node-version: "*"
+      - name: Install Yarn
+        run: corepack enable
+      - name: Install npm dependencies
+        run: yarn install
+      - name: Build Tauri
+        run: yarn run build:tauri
+      - name: Compress App
+        run: tar -cvzf qcvis.app.tar.gz ./src-tauri/target/release/bundle/macos/qcvis.app
```

Comment:
Doesn't `upload-artifact` itself perform compression?

Reply:
yes, but since `qcvis.app` is a folder we want to include, we wrap everything in a tar to attach it to the release

Reply:
I was still mostly thinking about the OS matrix in this case. The `release`-stage (which I want to omit right now) could do the final bundling

Reply:
I had problems, uploading the artifact, retaining the folder structure.

Reply:
Ok then leave it as is (with a conditional if you switch to an OS matrix)